src/Servant/JS/Vanilla.hs

```diff
@@ -34,7 +34,7 @@ generateVanillaJSWith opts req = "\n" <>
  <> "  xhr.open('" <> decodeUtf8 method <> "', " <> url <> ", true);\n"
  <>    reqheaders
  <> "  xhr.setRequestHeader('Accept', 'application/json');\n"
- <> (if isJust (req ^. reqBody) then "  xhr.setRequestHeader('Content-Type', 'application/json');\n" else "")
+ <> (if isJust (req ^. reqBody) && (req ^. reqBodyIsJSON)  then "  xhr.setRequestHeader('Content-Type', 'application/json');\n" else "")
```

Comment:
what about non-Vanilla drivers?

Reply:
I don't use them and didn't implement anything for them.